src/Essential.Diagnostics.RollingFileTraceListener/Diagnostics/RollingFileTraceListener.cs

```diff
@@ -111,6 +112,40 @@ public bool ConvertWriteToEvent
             }
         }
 
+        private bool? _newStreamOnError = null;
+        /// <summary>
+        /// Gets or sets whether errors writing to the file should cause a new file stream to be instantiated.
+        /// Useful when the drive containing the file has a transient fault (usb stick removed and reinserted, network outage of mapped drive, etc.)
+        /// </summary>
+        [SuppressMessage("Microsoft.Usage", "CA1806:DoNotIgnoreMethodResults", MessageId = "System.Boolean.TryParse(System.String,System.Boolean@)", Justification = "Default value is acceptable if conversion fails.")]
+        public bool NewStreamOnError
+        {
+            get
+            {
+                if (_newStreamOnError.HasValue)
+                {
+                    return _newStreamOnError.Value;
+                }
+
+                // Default behaviour is to create a new stream on error
+                var newStreamOnError = true;
+                if (Attributes.ContainsKey("newStreamOnError"))
+                {
+                    if(bool.TryParse(Attributes["newStreamOnError"], out newStreamOnError) == false)
```

Comment:
I don't think the if statement is required; the old code was fine; if it can't pass the string then it will stay at the value set above (true).

Reply:
In the MSDN docs on bool.TryParse it says 

> _ If the conversion failed, [result] contains false_

I took that to mean that the out value will be set to false and not keep its original value set to _true_ on the lines bove.